app/code/Magento/Sales/Model/Order/Invoice.php

```diff
@@ -236,7 +236,7 @@ public function setOrder(\Magento\Sales\Model\Order $order)
     public function getOrder()
     {
         if (!$this->_order instanceof \Magento\Sales\Model\Order) {
```

Comment:
Since `OrderRepository` may return any class implementing `OrderInterface` this check may be false-negative in some scenarios on customized system.